code/script.js

```diff
@@ -18,29 +19,154 @@ const showMessage = (message, sender) => {
+  if (!validNamePattern.test(name))  { 
+    showMessage("Invalid input. Please enter only letters and spaces.", 'bot');
+    nameInput.value = '';
+  } else  { 
+  showMessage(name, 'user');
+  nameInput.value = '';
+  // This renders the next question after 1 sec by invoking the next function. Taking the 'name' variable to the next function.
+  setTimeout(() => intro(name), 1000);
+  }
+};
+
+//This function displays a bot message in response to the handleNameInput function above and invokes the next question/function.
+const intro = (name) => {
+  showMessage (`Hi ${name}! Do you want to learn some Danish?`, 'bot');
+  setTimeout(showYesNobtn, 1000);
+}
+
+//This function invokes two buttons "Sure!" and "No thanks."
+const showYesNobtn = () => {
+  inputWrapper.innerHTML = `
+  <button id="yesBtn">Sure!</button>
+  <button id="noBtn">No thanks.</button>
+  `
+  document.getElementById('yesBtn').addEventListener('click', () => userReply('Sure!'));
+  document.getElementById('noBtn').addEventListener('click', () => userReply('No thanks.'));
+}
+
+//If the user clicks Sure! A new question will pop up after 1 sec. If the user clicks No thanks a goodbye message will appear after 1 sec and empty the input field.
+const userReply = (userReply) => {
+  if (userReply === 'Sure!') {
+    showMessage(`Sure!`, 'user')
+    setTimeout(() => presentWordOptions(), 1000);
+  } else if (userReply === 'No thanks.') {
+    showMessage(`No thanks.`, 'user')
+    setTimeout(sayGoodbye, 1000)
+  }
+}
+
+const sayGoodbye = () => {
+  showMessage (`Alright. See you another time!`, 'bot');
+  inputWrapper.innerHTML = ``;
+}
+
+//If the user clicks on "Sure!" this question will be asked and the user will be given three word options to chose from.
+const presentWordOptions = () => {
+  showMessage(`What word would you like to learn?`, "bot");
+  inputWrapper.innerHTML = `
+  <button id="sausage">Sausage</button>
+  <button id="milk">Milk</button>
+  <button id="butter">Butter</button>
+`;
+  document.getElementById('sausage').addEventListener('click', () => showSausageMessage());
+  document.getElementById('milk').addEventListener('click', () => showMilkMessage());
+  document.getElementById('butter').addEventListener('click', () => showButterMessage());
+}
+
+//This function will start if the user clicks on "Sausage". 
+const showSausageMessage = (selectedWord) => {
+  showMessage('Sausage!', 'user');
+  setTimeout(() => {
+    showMessage(`In Danish, it's called "pølse"`, 'bot');
+    setTimeout(() => askLearnAnotherWord(selectedWord), 1000);
+  }, 1000);
+}
+
+//This function will start if the user clicks on "Milk". 
+const showMilkMessage = (selectedWord) => {
+  showMessage('Milk!', 'user');
+//add comment
+  setTimeout(() => {
+    showMessage(`In Danish, it's called "melk".`, 'bot');
+    setTimeout(() => askLearnAnotherWord(selectedWord), 1000);
+  }, 1000);
```

Comment:
In some of your "setTimeout()" you have put '1000' twice, do they work differently from the ones where you only write it once?